testing/demo/pkg/setup/tendermint.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"os"
+	"os/exec"
+
+	"github.com/celestiaorg/celestia-zkevm-ibc-demo/testing/demo/pkg/utils"
+	"github.com/cosmos/solidity-ibc-eureka/abigen/ics26router"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// CreateTendermintLightClient creates the Tendermint light client on the EVM roll-up.
+func CreateTendermintLightClient() error {
+	err := utils.CheckSimappNodeHealth(simappRPC, 10)
+	if err != nil {
+		return fmt.Errorf("simapp node is not healthy, please ensure it is running correctly: %w", err)
+	}
+
+	err = utils.CheckEthereumNodeHealth(ethereumRPC)
+	if err != nil {
+		return fmt.Errorf("ethereum node is not healthy, please ensure it is running correctly: %w", err)
+	}
+
+	if err := deployEurekaContracts(); err != nil {
+		return err
+	}
+
+	addresses, err := utils.ExtractDeployedContractAddresses()
+	if err != nil {
+		return err
+	}
+
+	if err := addClientToRouter(addresses); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// deployEurekaContracts deploys all of the IBC Eureka contracts (including the
+// SP1 ICS07 Tendermint light client contract) on the EVM roll-up.
+func deployEurekaContracts() error {
```

Comment:
Q: How are the values for initialising specifically the sp1 tendermint light client contract set?

Reply:
They're read from `solidity-ibc-eureka/scripts/genesis.json` which we had a bunch of issues with in the past b/c the deployment commands previously would invoke a `just` command to generate that file and then modify it with verifier keys for the hollowed out SP1 programs.

Now the `makefile` in our repo creates the genesis.json directly using the `operator` binary. It sets the correct verifier keys.